jvm/hermes/src/main/jni/JJSIValue.h

```diff
@@ -148,21 +152,37 @@ class JJSIValue : public JJSIValueHybridClass {
     local_ref<JJSIObject_jhybridobject> asObject(alias_ref<JRuntimeThreadContext>, alias_ref<JJSIRuntime::jhybridobject> jRuntime);
     std::string toString(alias_ref<JRuntimeThreadContext>, alias_ref<JJSIRuntime::jhybridobject> jRuntime);
 
+    ~JJSIValue() override {
+        release();
+    }
+
     void release() override {
-        value_.reset();
+        if (!tracked && value_) value_.reset();
+        if (scope_) scope_->clearRef(this);
```

Comment:
_Ah_, I get why this is necessary - but maybe the `!tracked && value_` case could just make it's own `RuntimeScope` so that all the APIs still go through a scope and you don't need explicit logic to handle both cases.